stages/org.osbuild.yum.repos

```diff
@@ -0,0 +1,182 @@
+        "description": "URLs of a GPG key files that can be used for signing metadata and packages of this repository.",
+        "minItems": 1,
+        "items": {
+          "type": "string",
+          "minLength": 1
+        }
+      },
+      "metalink": {
+        "type": "string",
+        "description": "URL of a metalink for the repository.",
+        "minLength": 1
+      },
+      "mirrorlist": {
+        "type": "string",
+        "description": "URL of a mirrorlist for the repository.",
+        "minLength": 1
+      },
+      "module_hotfixes": {
+        "type": "boolean",
+        "description": "Set this to True to disable module RPM filtering and make all RPMs from the repository available."
+      },
+      "name": {
+        "type": "string",
+        "description": "A human-readable name of the repository. Defaults to the ID of the repository.",
+        "minLength": 1
+      },
+      "priority": {
+        "type": "integer",
+        "description": "The priority value of this repository."
+      },
+      "gpgcheck": {
+        "type": "boolean",
+        "description": "Whether to perform GPG signature check on packages found in this repository."
+      },
+      "repo_gpgcheck": {
+        "type": "boolean",
+        "description": "Whether to perform GPG signature check on this repository's metadata."
+      }
+    }
+  }
+},
+"additionalProperties": false,
+"description": "YUM / DNF repo file configuration.",
+"properties": {
+  "filename": {
+    "type": "string",
+    "pattern": "^[\\w.-]{1,250}\\.repo$",
+    "description": "Repo file name."
+  },
+  "repos": {
+    "type": "array",
+    "description": "YUM / DNF repo definitions.",
+    "minItems": 1,
+    "items": {
+        "$ref": "#/definitions/repo"
+    }
+  }
+}
+"""
+
+
+def option_value_to_str(value):
+    """
+    Convert allowed types of option values to string.
+
+    DNF allows string lists as a option value.
+    'dnf.conf' man page says:
+    "list   It is an option that could represent one or more strings separated by space or comma characters."
+    """
+    if isinstance(value, list):
+        value = " ".join(value)
+    elif isinstance(value, bool):
+        value = "1" if value else "0"
+    elif not isinstance(value, str):
+        value = str(value)
+    return value
+
+
+def main(tree, options):
+    filename = options.get("filename")
+    repos = options.get("repos")
+
+    yum_repos_dir = f"{tree}/etc/yum.repos.d"
+    os.makedirs(yum_repos_dir, exist_ok=True)
+
+    parser = configparser.ConfigParser()
+
+    for repo in repos:
+        repo_id = repo.pop("id")
+        parser.add_section(repo_id)
+        # Set some options in a specific order in which they tend to be
+        # written in repo files.
+        for option in [
+            "name",
+            "baseurl",
+            "metalink",
+            "mirrorlist",
+            "enabled",
+            "gpgcheck",
+            "repo_gpgcheck",
+            "gpgkey"
+        ]:
```

Comment:
bit hard to read, maybe extract it to a global variable `SUPPORTED_OPTIONS` or something?

Reply:
Sure, I can extract this to a variable. Nevertheless it is not a list of supported options, but list of options which should be listed in the repo configuration in this specific order if set in the stage options. My reasoning is that repo configurations as shipped by distributions or created by various tools (COPR, RHSM) tend to order some options in a specific way, therefore if we just iterated over the dictionary items, the order would be different than how are repository configurations usually structured (but it would be still valid).

Reply:
Makes sense! I was about to say that this is super helpful and should be in a comment next to the list but I see you have already done that! Perfect!